### DISCUSSION TITLE
Migration between kubevirt clusters - Source version is not supported

### DISCUSSION THREAD

**Discussion by jsakil14** [kubev2v/forklift#918] [Q&A]
Hello , I am back here after successfully installing and running a migration plan between Vmware & kubevirt cluster - Now , I am testing different use cases as below:
**1. Since , we would have vmtools installed in source machine , do we need to add the qemu agent for kubevirt side of things ? If so , do we go about adding hooks for uninstall of vmtools & install of qemu agent on the destination VM ? Any directions for this 
2. I was testing the inter-kubevirt cluster migrations , but was getting this error -SOURCE VERSION IS NOT FOUND - I believe it is checking for k8s version which I have listed below and as per requirements.** 

`{"level":"info","ts":"2024-05-27 14:18:24.953","logger":"plan|2qgnn","msg":"Condition added.","plan":{"name":"kubevirttovmware","namespace":"konveyor-forklift"},"condition"':`**{"type":"UnsupportedVersion","status":"True","reason":"NotSupported","category":"Critical","message":"Source version is not supported.","lastTransitionTime":null}}**
```
'{"level":"debug","ts":"2024-05-27 14:18:24.953","logger":"events","msg":"Map.Network does not have Ready condition.","type":"Warning","object":{"kind":"Plan","namespace":"konveyor-forklift","name":"kubevirttovmware","uid":"57c1580b-72c6-4cf4-9982-5fcb937174bf","apiVersion":"forklift.konveyor.io/v1beta1","resourceVersion":"80317515"},"reason":"NetworkMapNotReady"}
{"level":"debug","ts":"2024-05-27 14:18:24.953","logger":"events","msg":"VM has unmapped networks.","type":"Warning","object":{"kind":"Plan","namespace":"konveyor-forklift","name":"kubevirttovmware","uid":"57c1580b-72c6-4cf4-9982-5fcb937174bf","apiVersion":"forklift.konveyor.io/v1beta1","resourceVersion":"80317515"},"reason":"VMNetworksNotMapped"}
{"level":"debug","ts":"2024-05-27 14:18:24.953","logger":"events","msg":"VM has unmapped storage.","type":"Warning","object":{"kind":"Plan","namespace":"konveyor-forklift","name":"kubevirttovmware","uid":"57c1580b-72c6-4cf4-9982-5fcb937174bf","apiVersion":"forklift.konveyor.io/v1beta1","resourceVersion":"80317515"},"reason":"VMStorageNotMapped"}
```
**{"level":"debug","ts":"2024-05-27 14:18:24.953","logger":"events","msg":"Source version is not supported.",** `
"type":"Warning","object":{"kind":"Plan","namespace":"konveyor-forklift","name":"kubevirttovmware","uid":"57c1580b-72c6-4cf4-9982-5fcb937174bf","apiVersion":"forklift.konveyor.io/v1beta1","resourceVersion":"80317515"},"reason":"UnsupportedVersion"}
{"level":"info","ts":"2024-05-27 14:18:24.960","logger":"plan|2qgnn","msg":"Reconcile ended.","plan":{"name":"kubevirttovmware","namespace":"konveyor-forklift"},"reQ":0}
{"level":"info","ts":"2024-05-27 14:18:26.267","logger":"plan|g56wn","msg":"Reconcile started.","plan":{"name":"kubevirttovmware","namespace":"konveyor-forklift"}}
{"level":"error","ts":"2024-05-27 14:18:26.284","logger":"validator|ocp","msg":"Multus network not found.","plan":"konveyor-forklift/kubevirttovmware","stacktrace":"github.com/konveyor/forklift-controller/pkg/controller/plan/adapter/ocp.(*Validator).NetworksMapped\n\tpkg/controller/plan/adapter/ocp/validator.go:176\ngithub.com/konveyor/forklift-controller/pkg/controller/plan.
`

**Source:**

` k version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.25.10
Kustomize Version: v4.5.7
Server Version: v1.25.10`

**Destination:**

` k version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.25.10
Kustomize Version: v4.5.7
Server Version: v1.25.10`

**3. Is it possible to migrate the vms from kubevirt to vmware & do we have any work in progress for live migration in future with shared disks like DBA for quourum disks.**

**Comment by bennyz** [comment]
yes, the minimum version is 1.26

**Reply by jsakil14** [reply]
Thanks. Will upgrade and try . Could you cite the requirements page? 

Also , could you help me with this - **Is it possible to migrate the vms from kubevirt to vmware & do we have any work in progress for live migration in future with shared disks like DBA for quourum disks.**

Since , we would have vmtools installed in source machine , do we need to add the qemu agent for kubevirt side of things ? If so , do we go about adding hooks for uninstall of vmtools & install of qemu agent on the destination VM ? Any directions for this

**Reply by jsakil14** [reply]
> ###
> [ ](/bennyz)

You mean the kubevirt version upgrade or kuberenets? How do we correlate?

**Comment by jsakil14** [comment]
> yes, the minimum version is 1.26

So is the migration between kubevirt clusters supported or is there any requirement..i

**Reply by bennyz** [reply]
yes, the required kubernetes version is 1.26, this inferred from the Openshift version that matches the Openshift Virtualization version we require, which is 4.13, and Openshift 4.13 correlates to kubernetes 1.26
